t/01-log_context.t

```diff
@@ -0,0 +1,41 @@
+use strict;
+use warnings;
+
+use Test::More;
+
+use Log::Any qw($log);
+use Log::Any::Adapter;
+use Path::Tiny;
+use JSON::MaybeUTF8 qw(:v1);
+use Test::Exception;
+use Test::MockModule;
+
+my $file_log_message;
+my $json_log_file = Path::Tiny->tempfile();
+
+sub do_context_text {
+    my %args = @_;
+
+    # Remove this line to avoid empty string initialization
+    $file_log_message = '';
+    $json_log_file->remove;
+    Log::Any::Adapter->import('DERIV', $args{import_args}->%*);
+    $log->adapter->set_context($args{context});
+    $log->warn("This is a warn log");
+    $file_log_message = $json_log_file->exists ? $json_log_file->slurp : '';
+    chomp($file_log_message);
+    lives_ok { $file_log_message = decode_json_text($file_log_message) }
+    'log message is a valid json';
+    # Read the log message from the file  
+    is( $file_log_message->{correlation_id},     '1241421662',           "context ok" );
+    $log->adapter->clear_context;
+}
+
+do_context_text(
```

Comment:
do we need to use subtest for better reporting?
And should we use the better description of the test?
We should be able to understand the purpose of a test by reading the name/description and not looking at the implementation

Reply:
what do you mean by test description ? Adding Comments to test?

Reply:
not a comment. Every test should be named to reflect what you are testing. you should not go to the test implementation details to know what you are testing.  
for example  

In this case we are actually doing two tests:

subtest 'test to verify that correlation_id is present in the log message after set_context'
subtest 'test to verify that correlation_id is not present in the new log message after clear_context'

and you remove unnecessary comments